orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy

```diff
@@ -77,15 +77,45 @@ class OperationsController {
 
   @RequestMapping(value = "/orchestrate", method = RequestMethod.POST)
   Map<String, Object> orchestrate(@RequestBody Map pipeline, HttpServletResponse response) {
-    parsePipelineTrigger(executionRepository, buildService, pipeline)
-    Trigger trigger = objectMapper.convertValue(pipeline.trigger, Trigger)
+    if (pipeline.plan) {
+      parseAndValidatePipeline(pipeline)
+      log.info('Not starting pipeline (plan: true): {}', value("pipelineId", pipeline.id))
+      return pipeline
+    }
+
+    Throwable pipelineError = null
+    try {
+      parseAndValidatePipeline(pipeline)
+    } catch (Throwable t) {
+      pipelineError = t
+    }
+
+    def augmentedContext = [trigger: pipeline.trigger]
+    def processedPipeline = contextParameterProcessor.process(pipeline, augmentedContext, false)
 
-    boolean plan = pipeline.plan ?: false
-    for (PipelinePreprocessor preprocessor : (pipelinePreprocessors ?: [])) {
-      pipeline = preprocessor.process(pipeline)
+    if (pipelineError == null) {
+      startPipeline(processedPipeline)
+    } else {
+      markPipelineFailed(processedPipeline, pipelineError)
+      throw pipelineError
     }
+  }
+
+  private void parseAndValidatePipeline(Map pipeline) {
+    try {
+      parsePipelineTrigger(executionRepository, buildService, pipeline)
 
-    pipeline.trigger = trigger
+      for (PipelinePreprocessor preprocessor : (pipelinePreprocessors ?: [])) {
+        pipeline = preprocessor.process(pipeline)
+      }
+    } catch (Throwable t) {
+      throw t
```

Comment:
this catch & throw the same thing seems unnecessary?

Reply:
This is really just so we can have a `finally` block that inflates the trigger JSON into a `Trigger` object regardless of whether `parsePipelineTrigger` succeeds or fails.  (As even if it fails, we need the trigger to exist so we can properly log the error in `markPipelineFailed`.)  I'm open to a better way of achieving this, though, if you have any suggestions.

Reply:
You can still have the `finally` block ... I don't believe the behavior would be any different if you removed the `catch`?

Reply:
Right, I'd been mistaken in thinking that would silence all errors, but you're correct.